examples/dfem/minimal_surface.cpp

```diff
@@ -0,0 +1,518 @@
+      {
+         const auto invJ = inv(J);
+         const auto dudx = dudxi * invJ;
+         return tuple{coeff(dudx) * dudx * transpose(invJ) * det(J) * w};
+      }
+   };
+
+   // This is the derivative of the residual form with respect to the
+   // solution $u$.
+   //
+   // The inputs and outputs follow the same rules as the MFApply operator.
+   struct ManualDerivativeApply
+   {
+      MFEM_HOST_DEVICE inline
+      auto operator()(
+         const tensor<real_t, dim> &ddelta_udxi,
+         const tensor<real_t, dim> &dudxi,
+         const tensor<real_t, dim, dim> &J,
+         const real_t &w) const
+      {
+         const auto invJ = inv(J);
+         const auto dudx = dudxi * invJ;
+         const auto ddelta_udx = ddelta_udxi * invJ;
+
+         const auto c = coeff(dudx);
+         const auto term1 = c * ddelta_udx;
+         const auto term2 = c * c * c * dot(dudx, ddelta_udx) * dudx;
+
+         return tuple{(term1 - term2) * transpose(invJ) * det(J) * w};
+      }
+   };
+
+   // This class implements the Jacobian of the minimal surface operator. It
+   // mostly acts as a wrapper to retrieve the Jacobian and apply essential
+   // boundary conditions appropriately.
+   class MinimalSurfaceJacobian : public Operator
+   {
+   public:
+      MinimalSurfaceJacobian(const MinimalSurface *minsurface,
+                             const Vector &x) :
+         Operator(minsurface->Height()),
+         minsurface(minsurface),
+         z(minsurface->Height())
+      {
+         minsurface->u.SetFromTrueDofs(x);
+         auto mesh_nodes = static_cast<ParGridFunction*>
+                           (minsurface->H1.GetParMesh()->GetNodes());
+
+         // One can retrieve the derivative of a DifferentiableOperator wrt a
+         // field variable if the derivative has been requested during the
+         // DifferentiableOperator::AddDomainIntegrator call.
+         dres_du = minsurface->res->GetDerivative(SOLUTION_U, {&minsurface->u}, {mesh_nodes});
+      }
+
+      void Mult(const Vector &x, Vector &y) const override
+      {
+         z = x;
+         z.SetSubVector(minsurface->ess_tdofs, 0.0);
+
+         dres_du->Mult(z, y);
+
+         auto d_y = y.HostReadWrite();
+         const auto d_x = x.HostRead();
+         for (int i = 0; i < minsurface->ess_tdofs.Size(); i++)
+         {
+            d_y[minsurface->ess_tdofs[i]] = d_x[minsurface->ess_tdofs[i]];
+         }
+      }
+
+      // Pointer to the wrapped MinimalSurface operator
+      const MinimalSurface *minsurface = nullptr;
+
+      // Pointer to the DifferentiableOperator that computes the Jacobian
+      std::shared_ptr<DerivativeOperator> dres_du;
+
+      // Temporary vector
+      mutable Vector z;
+   };
+
+   // This class implements the Jacobian of the minimal surface operator using
+   // manually computed derivatives.
+   class MinimalSurfaceHandcodedJacobian : public Operator
+   {
+      static constexpr int DIRECTION_U = 3;
```

Comment:
What is a DIRECTION_U? Isn't it the increment of the solution?  Some description will be helpful here.

Reply:
added description